src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -0,0 +1,1885 @@
+			advancedControllableProperties.add(createButton(Command.FOCUS.getName() + HASH + FocusControl.NEAR.getName(), PLUS));
+		}
+	}
+
+	/**
+	 * This method is used for populate all AE control properties
+	 *
+	 * @param stats is the map that store all statistics
+	 * @param advancedControllableProperties is the list that store all controllable properties
+	 */
+	private void populateAEControl(Map<String, String> stats, List<AdvancedControllableProperty> advancedControllableProperties) {
+		stats.put(Command.EXPOSURE.getName() + HASH + Command.AE_MODE.getName(), "");
+
+		List<String> aeModeList = new ArrayList<>();
+		aeModeList.add(AEMode.FULL_AUTO.getName());
+		aeModeList.add(AEMode.IRIS_PRIORITY.getName());
+		aeModeList.add(AEMode.SHUTTER_PRIORITY.getName());
+		aeModeList.add(AEMode.MANUAL.getName());
+
+		AEMode aeMode = this.getAEMode();
+		assert aeMode != null;
+		advancedControllableProperties.add(createDropdown(Command.EXPOSURE.getName() + HASH + Command.AE_MODE.getName(), aeModeList, aeMode.getName()));
+
+		// Getting auto slow shutter status
+		SlowShutterStatus autoSlowShutterStatus = getAutoSlowShutterStatus();
+
+		switch (aeMode) {
+			case FULL_AUTO: {
+				// Populate backlight control
+				populateBacklightControl(stats, advancedControllableProperties);
+
+				// Populate exposure control
+				populateExposureControl(stats, advancedControllableProperties);
+
+				// Populate gain limit control
+				populateGainLimitControl(stats, advancedControllableProperties);
+
+				// Populate slow shutter control
+				assert autoSlowShutterStatus != null;
+				populateAutoSlowShutterControl(stats, advancedControllableProperties, autoSlowShutterStatus);
+				break;
+			}
+			case SHUTTER_PRIORITY: {
+				// Populate exposure control
+				populateExposureControl(stats, advancedControllableProperties);
+
+				// Populate gain limit control
+				populateGainLimitControl(stats, advancedControllableProperties);
+
+				// Populate shutter control
+				populateShutterControl(stats, advancedControllableProperties);
+				break;
+			}
+			case IRIS_PRIORITY: {
+				// Populate exposure control
+				populateExposureControl(stats, advancedControllableProperties);
+
+				// Populate gain limit control
+				populateGainLimitControl(stats, advancedControllableProperties);
+
+				// Populate slow shutter control
+				assert autoSlowShutterStatus != null;
+				populateAutoSlowShutterControl(stats, advancedControllableProperties, autoSlowShutterStatus);
+
+				// Populate iris control
+				populateIrisControl(stats, advancedControllableProperties);
+				break;
+			}
+			case MANUAL:
+				// Populate shutter control
+				populateShutterControl(stats, advancedControllableProperties);
+
+				// Populate gain control
+				populateGainControl(stats, advancedControllableProperties);
+
+				// Populate iris control
+				populateIrisControl(stats, advancedControllableProperties);
```

Comment:
no _break_ statement here, no _default_ case.

Reply:
thank you, I will fix it